core/src/main/java/com/linecorp/armeria/server/metric/PrometheusVersion1ExpositionService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.metric;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.AbstractHttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.TransientHttpService;
+import com.linecorp.armeria.server.TransientServiceOption;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufOutputStream;
+import io.prometheus.metrics.expositionformats.ExpositionFormatWriter;
+import io.prometheus.metrics.expositionformats.ExpositionFormats;
+import io.prometheus.metrics.model.registry.PrometheusRegistry;
+
+/**
+ * Exposes Prometheus metrics in <a href="https://prometheus.io/docs/instrumenting/exposition_formats/">
+ * EXPOSITION FORMATS</a>.
+ */
+public final class PrometheusVersion1ExpositionService extends AbstractHttpService
```

Comment:
Question) I'm curious if we want to use this opportunity to just create a separate `prometheus` integration module. The reasoning for this is:
- Users who are willing to upgrade won't be penalized by breaking changes twice. (Once from `PrometheusExpositionService` -> `PrometheusVersion1ExpositionService`, later from `PrometheusVersion1ExpositionService` to `PrometheusExpositionService` again)
- Upgrading may be easier if users can just plug in a different dependency instead of code-level changes
- We don't really use `prometheus` in our core logic

Reply:
Actually, I realized that this is more about `micrometer-prometheus` and not `prometheus` itself.
I think as-is is probably fine then since a separate module seems like overkill😅